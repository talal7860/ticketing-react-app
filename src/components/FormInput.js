import React from 'react';
import { FormGroup, Label, Input } from 'reactstrap';
import PropTypes from 'prop-types';

const FormInput = ({ id, label, name, type, placeholder, onChange, value }) => (
	<FormGroup>
		<Label for={id}>{label}</Label>
		<Input type={type} name={name} id={id} onChange={onChange} placeholder={placeholder} value={value} />
	</FormGroup>
);


FormInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
};

FormInput.defaultProps = {
  value: "",
};

export default FormInput;
