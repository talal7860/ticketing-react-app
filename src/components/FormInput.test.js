import React from 'react';
import { shallow } from 'enzyme';
import FormInput from './FormInput';
import { FormGroup } from 'reactstrap';

describe('FormInput', () => {
  it('should render a FormGroup tag when this input is rendered  provided', () => {
    const wrapper = shallow(<FormInput id="id" type="type" label="label" placeholder="placeholder" name="yo" value="" onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

});
