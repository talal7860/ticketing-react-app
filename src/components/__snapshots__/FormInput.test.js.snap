// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render a FormGroup tag when this input is rendered  provided 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormInput
    id="id"
    label="label"
    name="yo"
    onChange={[Function]}
    placeholder="placeholder"
    type="type"
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          for="id"
          tag="label"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }
>
          label
</Label>,
        <Input
          id="id"
          name="yo"
          onChange={[Function]}
          placeholder="placeholder"
          tag="p"
          type="type"
          value=""
/>,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "label",
          "for": "id",
          "tag": "label",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": "label",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id",
          "name": "yo",
          "onChange": [Function],
          "placeholder": "placeholder",
          "tag": "p",
          "type": "type",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            for="id"
            tag="label"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            label
</Label>,
          <Input
            id="id"
            name="yo"
            onChange={[Function]}
            placeholder="placeholder"
            tag="p"
            type="type"
            value=""
/>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "label",
            "for": "id",
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": "label",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id",
            "name": "yo",
            "onChange": [Function],
            "placeholder": "placeholder",
            "tag": "p",
            "type": "type",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
